completion/services.py

```diff
@@ -76,6 +76,28 @@ def fill_in_run(block_key):
                 completions[candidate] = 0.0
         return completions
 
+    def get_user_children(self, node):
+        """
+        Verticals will sometimes have children that also have children (some
+        examples are content libraries, split tests, conditionals, etc.). This
+        function will recurse through the children to get to the bottom leaf
+        and return only those children. The nodes passed in should have already
+        taken the user into account so the proper blocks are shown for this user.
+        """
+        if node.scope_ids.block_type == 'library_content':
+            node_children = node.get_child_descriptors()
```

Comment:
I think you can just use this method all the time, actually. You'll end up with the same thing as `get_children` in cases where the block doesn't have any dynamic children.

Reply:
Actually... so, I think `get_child_descriptors` only applies to `XModules`, not to pure `XBlocks`. But `get_children` is used by both. So, I'd use `get_child_descriptors` if it's available, and then `get_children` if it's not.

Reply:
Updated to prefer `get_child_descriptors` and fall back to `get_children`